File: app/src/main/java/com/example/bottlerocketstudioscodingtest/StoreAdapter.java
package com.example.bottlerocketstudioscodingtest;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.bottlerocketstudioscodingtest.Model.Store;

import java.util.ArrayList;

public class StoreAdapter extends RecyclerView.Adapter<StoreAdapter.ViewHolder>{
    private Context context;
    private ArrayList<Store> stores;

    public StoreAdapter(ArrayList<Store> stores) {
        this.stores = stores;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        context = viewGroup.getContext();
        View storeItem = LayoutInflater.from(context)
                .inflate(R.layout.store_item, viewGroup, false);
        return new StoreAdapter.ViewHolder(storeItem);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        final Store store = stores.get(i);

        String logo = store.getStoreLogoURL();
        String number = store.getPhone();
        String address = store.getAddress();
        String name = store.getName();

        viewHolder.mStorePhoneNumberView.setText(number);
        viewHolder.mStoreAddressView.setText(address);
        viewHolder.mStoreNameView.setText(name);

        Glide.with(context).load(logo).into(viewHolder.mImageView);

        viewHolder.itemView.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, DetailActivity.class);
                intent.putExtra("store", store);
                context.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return stores.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        private TextView mStoreAddressView;
        private TextView mStorePhoneNumberView;
        private TextView mStoreNameView;
        private ImageView mImageView;
        private CardView mCardView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            mStoreAddressView = itemView.findViewById(R.id.store_address);
            mStorePhoneNumberView = itemView.findViewById(R.id.store_phone);
            mStoreNameView = itemView.findViewById(R.id.store_name);
            mImageView = itemView.findViewById(R.id.store_logo);
        }
    }
}

File: app/src/main/java/com/example/bottlerocketstudioscodingtestkotlin/UI/StoreAdapter.kt
package com.example.bottlerocketstudioscodingtestkotlin.UI

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.example.bottlerocketstudioscodingtestkotlin.Model.Store
import com.example.bottlerocketstudioscodingtestkotlin.R

import java.util.ArrayList

class StoreAdapter(private val stores: ArrayList<Store>) : RecyclerView.Adapter<StoreAdapter.ViewHolder>() {
    private var context: Context? = null

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        context = viewGroup.context
        val storeItem = LayoutInflater.from(context)
                .inflate(R.layout.store_item, viewGroup, false)
        return ViewHolder(storeItem)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val store = stores[i]

        val logo = store.storeLogoURL
        val number = store.phone
        val address = store.address
        val name = store.name
        val city = store.city
        val id = store.storeID

        viewHolder.mStorePhoneNumberView.setText(R.string.phone_label)
        viewHolder.mStorePhoneNumberView.append(" $number")
        viewHolder.mStoreAddressView.text = address
        viewHolder.mStoreAddressView.append(", $city")
        viewHolder.mStoreNameView.text = name

        Glide.with(context!!).load(logo).into(viewHolder.mImageView)

        viewHolder.itemView.setOnClickListener {
            val intent = Intent(context, DetailActivity::class.java)
            intent.putExtra("store", store)
            context!!.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return stores.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mStoreAddressView: TextView
        val mStorePhoneNumberView: TextView
        val mStoreNameView: TextView
        val mImageView: ImageView

        init {
            mStoreAddressView = itemView.findViewById(R.id.store_address)
            mStorePhoneNumberView = itemView.findViewById(R.id.store_phone)
            mStoreNameView = itemView.findViewById(R.id.store_name)
            mImageView = itemView.findViewById(R.id.store_logo)
        }
    }
}
